synapse/storage/databases/main/roommember.py

```diff
@@ -1013,12 +1013,14 @@ async def get_current_hosts_in_room_ordered(self, room_id: str) -> List[str]:
             # `get_users_in_room` rather than funky SQL.
 
             domains = await self.get_current_hosts_in_room(room_id)
-            return list(domains)
+            return tuple(domains)
```

Comment:
Hmmmmmm.

This function is supposed to return servers ordered by how long they've been in the room. But `get_current_hosts_in_room` returns an AbstractSet which has no particular order. Did we break this recently?

Reply:
Oh, from the docstring:

>  For SQLite the returned list is not ordered, as SQLite doesn't support
the appropriate SQL.